tests/test_routes.py

```diff
@@ -34,20 +35,63 @@ def test_download_get(requests_mock, flask_test_client):
     assert page.select_one(".govuk-back-link") is None
 
 
-@pytest.mark.usefixtures("mock_get_response_json")
-def test_download_post_json(flask_test_client):
-    response = flask_test_client.post("/download", data={"file_format": "json"})
-    assert response.status_code == 200
-    assert response.mimetype == "application/json"
-    assert response.data == b'{"data": "test"}'
+def test_process_async_download_call(flask_test_client, mocked_routes_process_async_download):
+    """Test that the download route calls the process_async_download function with the correct parameters,
+    including the user email address."""
+
+    flask_test_client.post("/download", data={"file_format": "xlsx"})
+    assert mocked_routes_process_async_download.called
+    assert mocked_routes_process_async_download.call_args.args[0] == {
+        "file_format": "xlsx",
+        "email_address": "test-user@example.com",
+    }
+
+
+def test_process_async_download_function(flask_test_client, mocked_download_data_process_async_download):
+    """Test that app/main/process_async_download() function returns status code 204 when
+    the download request is successful."""
 
+    status_code = process_async_download({"file_format": "xlsx", "email_address": "test-user@example.com"})
+    assert status_code == 204
+
+
+def test_async_download_redirect_OK(flask_test_client, mocked_download_data_process_async_download):
+    """Test that the download route redirects to the request-received page after a successful download request."""
 
-@pytest.mark.usefixtures("mock_get_response_xlsx")
-def test_download_post_xlsx(flask_test_client):
     response = flask_test_client.post("/download", data={"file_format": "xlsx"})
-    assert response.status_code == 200
-    assert response.mimetype == "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
-    assert response.data == b"xlsx data"
+    assert response.status_code == 302  # redirect to request-received page
+    assert response.headers["Content-Type"] == "text/html; charset=utf-8"
+    expected_location = url_for("main.request_received", _external=True)
+    assert (
+        response.headers["Location"] in expected_location
+    )  # /request-received in http://localhost:4002/request-received'
```

Comment:
```suggestion
    expected_location = url_for("main.request_received", _external=False)
    assert (
        response.headers["Location"] == expected_location
    )
```

I think if you make this an internal URL, you can do a straight comparison, which feels stricter?

Reply:
Changed to internal url and stricter comparison.